src/polygon.jl

```diff
@@ -0,0 +1,508 @@
+                img[i, j_left:j_right] .= triangle.color # Draw the line between the 2 points you found
+                # We do this in the if statement because we don't want to draw a line if no point of the triangle was found on this line
+
+            end
+            j -= 1
+        end
+
+    end
+
+    return img
+end
+
+"""
+    drawimage(triangles::Array, canvas::Array)
+
+Goes over all triangles in an array and draws them all on the same canvas, then returns the resulting image. 
+"""
+function drawimage(triangles::Array, canvas::Array)
+    polyimg = copy(canvas)
+    for triangle in triangles
+        polyimg = drawtriangle(triangle, polyimg)
+    end
+    return polyimg
+end
+
+"""
+    colordiffsum(img1::Array, img2::Array, m::Int, n::Int)
+
+The objective function of the algorithm. Computes the difference between 2 images by computing the difference in color for every pixel.
+"""
+colordiffsum(img1::Array, img2::Array) = sum(colordiff.(img1, img2))
+
+"""
+    checktriangle(triangle::Triangle, m::Int, n::Int)
+
+Checks a triangle to make sure we don't have any STUPID triangles.
+A stupid triangle is a triangle which:
+- Is much too thin (a couple of pixels thin)
+- Has all 3 points outside of the canvas
+- Is very stretched out
+These stupid triangles can all mess with the drawtriangle function, which is why checking for them and weeding them out is essential.
+Otherwise we get errors and boy do we hate errors.
+"""
+function checktriangle(triangle::Triangle, m::Int, n::Int)
+    # Rather than using the actual boundaries of the canvas, we use a slightly smaller rectangle for safety reasons
+    s = 10 # safety factor
+    x1, x2, x3 = real(triangle.p1), real(triangle.p2), real(triangle.p3)
+    y1, y2, y3 = imag(triangle.p1), imag(triangle.p2), imag(triangle.p3)
+
+    max_y = max(y1, y2, y3)
+    min_y = min(y1, y2, y3)
+    max_delta_y = max_y - min_y
+    YisStupid = max_delta_y <= s && return true # Too thin
+
+    max_x = max(x1, x2, x3)
+    min_x = min(x1, x2, x3)
+    max_delta_x = max_x - min_x
+    XisStupid = max_delta_x <= s && return true # Too thin
+
+    p1_out_of_canvas = (x1 <= s || x1 >= (n - s)) || (y1 <= s || y1 >= (m - s))
+    p2_out_of_canvas = (x2 <= s || x2 >= (n - s)) || (y2 <= s || y2 >= (m - s))
+    p3_out_of_canvas = (x3 <= s || x3 >= (n - s)) || (y3 <= s || y3 >= (m - s))
+    pointsAreStupid = all([p1_out_of_canvas, p2_out_of_canvas, p3_out_of_canvas]) && return true # All points are (almost) outside of the canvas, that's no bueno
+
+    # Very elongated triangles cause trouble: check for these as well
+    # Credit for idea of using the centroid: Her Magnificence, Gender Equality Expert Dr. Roets
+    midpoint = (triangle.p1 + triangle.p2 + triangle.p3)/3
+    dist1 = abs(triangle.p1 - midpoint)
+    dist2 = abs(triangle.p2 - midpoint)
+    dist3 = abs(triangle.p3 - midpoint)
+    isStretchyBoi = max(dist1, dist2, dist3) > 2*min(dist1, dist2, dist3) && return true
```

Comment:
I appreciate your sense of humour, and I think good jokes are absolutely necessary in any team. However, the code should be as unambiguous as possible. Some people might not get what you're talking about(and you have to add comments, which further pollutes the code) or be in a wrong mood. I'd suggest renaming `isStretchyBoi` to `iselongated`.